patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -86,7 +86,8 @@ internal static LocalMod[] FindAllMods()
 
 		// Workshop folders
 		WorkshopFileFinder.Refresh(new WorkshopIssueReporter());
-		foreach (string repo in WorkshopFileFinder.ModPaths) {
+		List<string> workshopModPaths = [.. WorkshopFileFinder.ModPaths]; // Avoid collection was modified
```

Comment:
This snapshotting should be hidden behind the `ModPaths` property, and combined with a `lock`

Reply:
On looking closer, it seems that `LocalModsChanged` is incredibly not thread-safe already. We should probably make sure that all of the callbacks registered to `OnLocalModsChanged` are using either `lock` (for actually cross thread-updates) or `QueueMainThreadAction` (for UI) properly